_posts/2023-10-04-migrating-postgres-via-replication.md

```diff
@@ -7,47 +7,29 @@ layout: default
 
 ### Options & Decision
 
 We were going to build a copy of our production infrastructure on Postgres 10, then carefully try out different migration strategies to move fully from 10 to 15. This turned out to be difficult. All the versions were so old - we would have needed EOL Ubuntu servers to build the EOL Postgres versions, which seemed like wasted effort.
 
-The first thing we did was we tried to do a dump and restore to see how long it took. It took 15 hours to restore. It took 45 minutes to back up. Setting that up also broke our regular backups, though we didn't know that yet.
-
-If we did a restore from backup we'd probably also have to reindex, which would take another 18 hours.
+The first thing we did was try a dump and restore to see how long it took. This is how we successfully migrated all our other databases. For Figgy, though, the dump took 45 minutes and restoring took 15 hours. Reindexing would have taken another 18 hours or more. If we had started the process on a Tuesday, we could have started the reindex on Wednesday and Figgy could have been back up on Thursday. But that would still be 2 days with the site in read-only mode. We abandoned this approach before we even reindexed. 
 
 We didn't want the migration to involve that much downtime and disruption. We also have and support a culture of not working after hours or on the weekend.
 
+Meanwhile, setting up that dump-and-restore process also broke our regular backups, though we didn't know that yet (dun, dun, dunnnnnnnn!).
+
 ### The Peril
 
-While we were discussing what to do next, our old infrastructure went down and took the site down with it.
+While we were discussing what to do next, our old infrastructure went down and took the site down with it. We knew how long dump-and-restore would take, and now we were faced with a unplanned outage - two days of downtime with no access to digital content.
 
-At this point we realized that the dump/restore we had done before had broken backups! The most recent backup we could find was six weeks old. Restoring from that backup did not go well. Importing a .sql file was going to take too long, so we tried a different approach. While we were working on that, the server came back up. We fixed the backups but we were feeling uncomfortable.
+This is when we realized that our attempt to dump and restore had broken backups! The most recent backup we could find was six weeks old. Restoring from that backup did not go well. Importing a .sql file was going to take too long, so we tried a different approach. While we were working on that, the server came back up. We fixed the backups but we were feeling even less comfortable than before.
 
 #### The Peril 2: Electric Boogaloo
```

Comment:
Having personally never seen "breakin'" or "breakin' 2" I humbly offer the following sequel title alternatives:

- "The New Batch" (Gremlins)
- "Peril Harder" (variation on the Die Hard sequel)
- "Freddy's Revenge" (Nightmare on Elm Street)